src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -177,6 +177,25 @@ StateBackend getStateBackend() {
     return stateBackend;
   }
 
+  @Override
+  public synchronized void reloadStateFromBackend() throws IOException {
+    GlobalStateInfo newGlobalStateInfo = getStateBackend().loadOrCreateGlobalState();
+    Map<String, IndexStateManager> newManagerMap = new HashMap<>();
+    for (Map.Entry<String, IndexGlobalState> entry :
+        newGlobalStateInfo.getIndicesMap().entrySet()) {
+      String indexName = entry.getKey();
+      IndexStateManager stateManager =
+          createIndexStateManager(indexName, entry.getValue().getId(), stateBackend);
+      stateManager.load();
+      newManagerMap.put(indexName, stateManager);
+    }
+    ImmutableState newImmutableState = new ImmutableState(newGlobalStateInfo, newManagerMap);
+    setImmutableState(newImmutableState);
```

Comment:
Setting the new `ImmutableState` should be done after starting any new indices. If there is a problem for some reason, the valid previous state will remain active.